src/witness/memory.rs

```diff
@@ -120,15 +138,16 @@ impl SafeMemory {
 
     /// Reads a Field Element from the memory at the specified offset
     pub fn read_fr(&self, ptr: usize) -> Result<BigInt> {
-        let view = self.memory.view::<u8>();
+        let store = self.store.read().unwrap();
+        let view = self.memory.view(&*store);
 
-        let res = if view[ptr + 4 + 3].get() & 0x80 != 0 {
+        let res = if view.read_u8(ptr as u64 + 4 + 3).unwrap() & 0x80 != 0 {
```

Comment:
is there a reason this function returns a `Result<BigInt>` as opposed to just `BigInt`? It seems that errors are not handled nor propagated

Reply:
Nice, I will update these to propagate the errors. The function previously returned `Result<BigInt>`, so I just left the signature the same. It looks like the callers of it do propagate the error though (although only in the circom1 calculation). Perhaps we remove this function all together when we remove circom1.